automat/_visualize.py

```diff
@@ -1,30 +1,70 @@
+import graphviz
 
-from __future__ import unicode_literals
 
 def _gvquote(s):
     return '"{}"'.format(s.replace('"', r'\"'))
 
+
 def _gvhtml(s):
     return '<{}>'.format(s)
 
 
-def graphviz(automaton, inputAsString=repr,
-             outputAsString=repr,
-             stateAsString=repr):
+def elementMaker(name, *children, **attrs):
+    """
+    Construct a string from the HTML element description.
     """
-    Produce a graphviz dot file as an iterable of bytes.
+    formattedAttrs = ' '.join('{}={}'.format(key, _gvquote(str(value)))
+                              for key, value in sorted(attrs.items()))
+    formattedChildren = ''.join(children)
+    return u'<{name} {attrs}>{children}</{name}>'.format(
+        name=name,
+        attrs=formattedAttrs,
+        children=formattedChildren)
 
-    Use like so::
 
-        with f as open('something', 'wb'):
-            f.writelines(graphviz(automaton))
+def tableMaker(inputLabel, outputLabels, port, _E=elementMaker):
+    """
+    Construct an HTML table to label a state transition.
     """
-    yield "digraph {\n"
-    # yield "pad=0.25;\n"
-    # yield 'splines="curved";\n'
-    yield "graph[pack=true,dpi=100]\n"
-    yield 'node[fontname="Menlo"]\n'
-    yield 'edge[fontname="Menlo"]\n'
+    colspan = {}
+    if outputLabels:
+        colspan['colspan'] = str(len(outputLabels))
+
+    inputLabelCell = _E("td",
+                        _E("font",
+                           inputLabel,
+                           face="menlo-italic"),
```

Comment:
This is for a follow-up PR for sure, but "menlo" is only going to be available on Macs, so we should probably make this customizable (and have a platform-dependent default…)